### PR TITLE
rgb to hsv 

### PR DESCRIPTION
**Pull request by dvd42**

### CODE REVIEW

File: torchgeometry/image/color.py [hunk 1]
```diff
@@ -1,21 +1,178 @@
...
+    # avoid division by zero
+    deltac = torch.where(deltac == 0, torch.ones_like(deltac), deltac)
+
+    rc = (maxc - r) / deltac
+    gc = (maxc - g) / deltac
+    bc = (maxc - b) / deltac
+
+    h = 4.0 + gc - rc
+    h[g == maxc] = 2.0 + rc[g == maxc] - bc[g == maxc]
+    h[r == maxc] = bc[r == maxc] - gc[r == maxc]
```
Comment (edgarriba) [inline torchgeometry/image/color.py]: create an auxiliary variable for g_max and r_max to avoid computing three time the same mask

File: torchgeometry/image/color.py [hunk 2]
```diff
@@ -1,21 +1,178 @@
...
+    def __init__(self) -> None:
+        super(RgbToHsV, self).__init__()
+
+    def forward(self, image: torch.Tensor) -> torch.Tensor:  # type: ignore
+        return rgb_to_hsv(image)
+
+
+def rgb_to_hsv(image):
+
+    """Convert an RGB image to HSV
```
Comment (edgarriba) [inline torchgeometry/image/color.py]: end definition with a point

File: torchgeometry/image/color.py [hunk 3]
```diff
@@ -1,21 +1,178 @@
...
+
+    if not torch.is_tensor(image):
+        raise TypeError("Input type is not a torch.Tensor. Got {}".format(
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W).Got {}"
+                         .format(image.shape))
+
+    out = image.flip(-3)
```
Comment (edgarriba) [inline torchgeometry/image/color.py]: could you put a comment here just saying that by doing this we flip the channels ?

File: torchgeometry/image/color.py [hunk 4]
```diff
@@ -1,21 +1,178 @@
...
+        input (torch.tensor): image to be converted to grayscale.
+
+    returns:
+        torch.tensor: grayscale version of the image.
 
-    Shape:
-        - Input: :math:`(*, 3, H, W)`
-        - Output: :math:`(*, 1, H, W)`
+    shape:
+        - input: :math:`(*, 3, h, w)`
```
Comment (edgarriba) [inline torchgeometry/image/color.py]: keep uppcase HW

File: torchgeometry/image/color.py [hunk 5]
```diff
@@ -1,21 +1,178 @@
...
+    return out
+
+
+class RgbToGrayscale(nn.Module):
+    r"""convert image to grayscale version of image.
+
+    the image data is assumed to be in the range of (0, 1).
+
+    args:
+        input (torch.tensor): image to be converted to grayscale.
```
Comment (edgarriba) [inline torchgeometry/image/color.py]: for convenience, put torch.Tensor

File: torchgeometry/image/color.py [hunk 6]
```diff
@@ -1,21 +1,178 @@
 import torch
 import torch.nn as nn
+import numpy as np
 
 
-class RgbToGrayscale(nn.Module):
-    r"""Convert image to grayscale version of image.
+class RgbToHsV(nn.Module):
```
Comment (edgarriba) [inline torchgeometry/image/color.py]: rename to `RgbToHsv`

File: torchgeometry/image/color.py [hunk 7]
```diff
@@ -1,21 +1,178 @@
...
 
+    """convert image from RGB to HSV.
```
Comment (edgarriba) [inline torchgeometry/image/color.py]: Convert

File: torchgeometry/image/color.py [hunk 8]
```diff
@@ -1,21 +1,178 @@
...
     The image data is assumed to be in the range of (0, 1).
 
+    args:
+        image (torch.tensor): RGB image to be converted to HSV.
```
Comment (edgarriba) [inline torchgeometry/image/color.py]: torch.Tensor

File: torchgeometry/image/color.py [hunk 9]
```diff
@@ -1,21 +1,178 @@
...
+
+    if not torch.is_tensor(image):
+        raise TypeError("Input type is not a torch.Tensor. Got {}".format(
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
+                         .format(image.shape))
+
+    r = image[..., 0, :, :]
```
Comment (edgarriba) [inline torchgeometry/image/color.py]: using typing to help functions tracing:
`r: torch.Tensor = image[..., 0, :, :]`
